README.md

```diff
@@ -2,4 +2,88 @@
 
 ## hadoop-azure-token-providers
 
-TODO
\ No newline at end of file
+The [`hadoop-azure` library](https://hadoop.apache.org/docs/stable/hadoop-azure/abfs.html) supports supplying a [custom token provider](https://hadoop.apache.org/docs/stable/hadoop-azure/abfs.html#Custom_OAuth_2.0_Token_Provider) for getting accessing tokens for Azure Storage.
+
+This package contains implementations of custom token providers (currently only `AzureCliCredentialTokenProvider`).
+
+### Development
+
+Open the repository in Visual Studio Code and use the provided Dev Container, which has the necessary Java SDK and Maven tools installed.
+
+#### Building the package
+
+Use Maven to build the package:
+
+`mvn clean package`
+
+#### Deploying the package
+
+The package is currently configured to deploy to [the public `hadoop` Artifacts feed](https://dev.azure.com/endjin-labs/hadoop/_artifacts/feed/hadoop) in the `endjin-labs` Azure DevOps instance as `com.endjin.hadoop:hadoop-azure-token-providers`.
+
+You need to provide credentials in a settings file to deploy to the feed. Create a `settings.xml` file in the `~/.m2` folder with the following contents:
+
+```xml
+<settings>
+  <servers>
+    <server>
+      <id>hadoop</id>
+      <username>endjin-labs</username>
+      <password>{Personal Access Token}</password>
+    </server>
+  </servers>
+</settings>
+```
+
+Replace `{Personal Access Token}` with a Personal Access Token generated from the `endjin-labs` Azure DevOps instance with permission to read/write Packages.
+
+In the `pom.xml` file, set the desired version number in the `<version>` tag, then 
+use Maven to deploy the package:
+
+`mvn deploy`
+
+> TODO: Add GitHub Actions workflow for automatically versioning & deploying the package
+
+### Usage from PySpark
+
+The package was built in order to enable a simple way of accessing data in Azure Data Lake when developing a PySpark solution locally in a Dev Container.
```

Comment:
This feels like it should be at the top of the file... and in the github project description